app/services/client_login_service.rb

```diff
@@ -0,0 +1,64 @@
+class ClientLoginService
+  class << self
+    def clients_for_token(raw_token)
+      # these might have multiple email addresses
+      to_addresses = EmailAccessToken.lookup(raw_token).pluck(:email_address)
+      emails = to_addresses.map { |to| to.split(",") }.flatten(1)
+      email_intake_matches = accessible_intakes.where(email_address: emails)
+      spouse_email_intake_matches = accessible_intakes.where(spouse_email_address: emails)
+      phone_numbers = TextMessageAccessToken.lookup(raw_token).pluck(:sms_phone_number)
+      phone_intake_matches = accessible_intakes.where(sms_phone_number: phone_numbers)
+      intake_matches = email_intake_matches.or(spouse_email_intake_matches).or(phone_intake_matches)
+
+      # Client.by_raw_login_token supports login links generated from mid-Jan through early March 2021.
+      Client.where(intake: intake_matches).or(Client.by_raw_login_token(raw_token))
+    end
+
+    def handle_email_request(email_address:, visitor_id:, locale: :en)
+      if can_login_by_email_verification?(email_address)
```

Comment:
Can this be in a bg job, so that timing doesn't reveal if the email exists with us?

Then the `perform_later` here can be `.perform_now` too I guess :D